src/Runner.Worker/JobExtension.cs

```diff
@@ -141,6 +141,10 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
                         context.Output($"Fail to parse and display GITHUB_TOKEN permissions list: {ex.Message}");
                         Trace.Error(ex);
                     }
+                    
+                    var secretSource = context.GetGitHubContext("secret_source");
+                    ArgUtil.NotNull(secretSource, nameof(secretSource));
```

Comment:
We better not to throw, since customers can use the latest runner with older version GHES which doesn't have this context populated.